jerry-core/ecma/base/ecma-globals.h

```diff
@@ -204,20 +204,99 @@ typedef enum
 } ecma_property_configurable_value_t;
 
 /**
- * Property's flag list.
+ * Property list:
+ *   The property list of an object is a chain list of various items.
+ *   The type of each item is stored in the first byte of the item.
+ *
+ *   The most common item is the property pair, which contains two
+ *   ecmascript properties. It is also important, that after the
+ *   first property pair, only property pair items are allowed.
+ *
+ *   Example for other items is property name hash map, or array of items.
+ */
+
+/**
+ * Property type list.
  */
 typedef enum
 {
-  ECMA_PROPERTY_FLAG_NAMEDDATA = 1u << 0, /**< property is named data */
-  ECMA_PROPERTY_FLAG_NAMEDACCESSOR = 1u << 1, /**< property is named accessor */
-  ECMA_PROPERTY_FLAG_INTERNAL = 1u << 2, /**< property is internal property */
-  ECMA_PROPERTY_FLAG_CONFIGURABLE = 1u << 3, /**< property is configurable */
-  ECMA_PROPERTY_FLAG_ENUMERABLE = 1u << 4, /**< property is enumerable */
-  ECMA_PROPERTY_FLAG_WRITABLE = 1u << 5, /**< property is writable */
-  ECMA_PROPERTY_FLAG_LCACHED = 1u << 6, /**< property is lcached */
+  ECMA_PROPERTY_TYPE_DELETED, /**< deleted property */
+  ECMA_PROPERTY_TYPE_INTERNAL, /**< internal property */
+  ECMA_PROPERTY_TYPE_NAMEDDATA, /**< property is named data */
+  ECMA_PROPERTY_TYPE_NAMEDACCESSOR, /**< property is named accessor */
+
+  ECMA_PROPERTY_TYPE_HASH, /**< property hash (TODO: implement this) */
```

Comment:
Right now, this enum label (`ECMA_PROPERTY_TYPE_HASH`) is not used anywhere. (Except for the `__MAX` label.) Please, remove this from the PR and add it back in another PR once there is an implementation that makes use of it.